src/BatchRequest.js

```diff
@@ -0,0 +1,165 @@
+'use strict'
+
+const HttpRequestBase = require('./HttpRequestBase'),
+      rp = require('request-promise-native'),
+      StandardError = require('@unplgtc/standard-error');
+
+let executionTimer;
+
+const BatchRequest = {
+	requests: [], // How do we clean these once they've executed?
+
+	stallMs: 50,
+	
+	stall(milliseconds) {
+		this.parent
+		? (this.parent.stallMs = milliseconds)
+		: (this.stallMs = milliseconds);
+
+		return this;
+	},
+
+	throttle(milliseconds) {
+		this.parent
+		? (this.parent.throttleMs = milliseconds)
+		: (this.throttleMs = milliseconds);
+
+		return this;
+	},
+
+	addRequest(cleanup) {
+		const child = this.spawnChild();
+		child.result = new Promise((resolve, reject) => {
+			child.resolve = resolve;
+			child.reject = reject;
+		});
+
+		this.requests.push(child);
+		this.cleanup = cleanup;
+
+		return child;
+	},
+
+	spawnChild() {
+		return Object.create(this)
```

Comment:
Great idea, will add this. BatchRequests should never be more than 1 level deep. `BatchRequest.addRequest()` is not expected to be a function that users ever call. To get a new BatchRequest you should call `HttpRequest.batch()` which will handle giving back the correct batch. Good catch though because that's not obvious nor protected against. Maybe I should change the non-user-facing functions to have underscores in front of their names to reinforce the convention?

Reply:
👍 I think underscores for the internal methods would be a good idea. Don't need to be obfuscated or anything like that, just underscores.

Reply:
Cool, I'll add that. For a while I've been playing with having internal methods packaged in a a different object that gets prototype-linked, but with the mild potential controversy around prototype linking I've been leaning toward just combining things and only using that pattern when it's actually necessary (Like the new HttpRequestBase, which is actually legitimately useful here)

Reply:
Unexpectedly need to get this merged asap so going to add this in an additional MR.